flang/lib/Optimizer/Transforms/OMPMapInfoFinalization.cpp

```diff
@@ -176,8 +204,9 @@ class OMPMapInfoFinalizationPass
                 mlir::IntegerType::get(builder.getContext(), 32)),
             llvm::ArrayRef<int32_t>({0})),
         /*bounds=*/mlir::SmallVector<mlir::Value>{},
-        builder.getIntegerAttr(builder.getIntegerType(64, false),
-                               op.getMapType().value()),
+        builder.getIntegerAttr(
+            builder.getIntegerType(64, false),
+            getDescriptorMapType(op.getMapType().value(), target)),
```

Comment:
Do we know for sure the `map_type` argument is defined by this point or would it make sense to do something like this instead?
```suggestion
            getDescriptorMapType(op.getMapType().value_or(llvm::omp::OpenMPOffloadMappingFlags::OMP_MAP_NONE), target)),
```

Reply:
Actually, maybe we could just make the `map_type` argument required but default-valued to 0 in MLIR instead.

Reply:
The map type should never be unset by Flang, which is the only frontend that will use this pass!